blockly-core/core/code_generation/generator.js

```diff
@@ -197,6 +197,11 @@ Blockly.CodeGenerator.prototype.blockToCode = function(block, opt_showHidden) {
   if (!block) {
     return '';
   }
+  if (block.skipGeneration) {
+    // Clear the flag, so it doesn't persist to future serializations.
+    block.skipGeneration = false;
```

Comment:
Ah, good call, this feels way cleaner!  Fixed in 3cf07ee7b5acfad5d77d70485ccb7454220c89b8.